pages/docs/reference/whatsnew14.md

```diff
@@ -6,6 +6,293 @@ title: "What's New in Kotlin 1.4"
+
+In Kotlin 1.3, you often needed to add explicit casts (either `!!` or type casts like `as String`) to make such cases work, 
+and now these casts have become unnecessary.
+
+#### Smart casts for callable references
+
+In Kotlin 1.3, you couldn’t access a member reference of a smart cast type. Now you can:
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+
+```kotlin
+import kotlin.reflect.KFunction
+
+sealed class Animal
+class Cat : Animal() {
+    fun meow() {
+        println("meow")
+    }
+}
+
+class Dog : Animal() {
+    fun woof() {
+        println("woof")
+    }
+}
+
+//sampleStart
+fun perform(animal: Animal) {
+    val kFunction: KFunction<*> = when (animal) {
+        is Cat -> animal::meow
+        is Dog -> animal::woof
+    }
+    kFunction.call()
+}
+//sampleEnd
+
+fun main() {
+    perform(Cat())
+}
+```
+
+</div>
+
+You can use different member references `animal::meow` and `animal::woof` after the animal variable has been smart cast 
+to specific types `Cat` and `Dog`. After type checks, you can access member references corresponding to subtypes.
+
+#### Better inference for delegated properties
+
+The type of a delegated property wasn’t taken into account while analyzing the delegate expression which follows the `by` 
+keyword. For instance, the following code didn’t compile before, but now the compiler correctly infers the types of the 
+`old` and `new` parameters as `String?`:
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+
+```kotlin
+import kotlin.properties.Delegates
+
+fun main() {
+    var prop: String? by Delegates.observable(null) { p, old, new ->
+        println("$old → $new")
+    }
+    prop = "abc"
+    prop = "xyz"
+}
+```
+
+</div>
+
+#### SAM conversion for Java interfaces with different arguments
+
+Kotlin has supported SAM conversions for Java interfaces from the beginning, but there was one case that wasn’t supported, 
+which was sometimes annoying when working with existing Java libraries. If you called a Java method that took two SAM interfaces 
+as parameters, both arguments need to be either lambdas or regular objects. You couldn't pass one argument as a lambda and 
+another as an object. 
+
+The new algorithm fixes this issue, and you can pass a lambda instead of a SAM interface in any case, 
+which is the way you’d naturally expect it to work.
+
+<div class="sample" markdown="1" theme="idea" mode="java" data-highlight-only>
+
+```java
+// FILE: A.java
+public class A {
+    public void static foo(Runnable r1, Runnable r2) {}
+}
+```
+
+</div>
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+// FILE: test.kt
+fun test(r1: Runnable) {
+    A.foo(r1) {}  // Works now
+}
+```
+
+</div>
+
+#### Java SAM interfaces in Kotlin
+
+Starting from Kotlin 1.4, you can use Java SAM interfaces in Kotlin. Previously, you could declare such a function `foo`
+only in Java code.
```

Comment:
Not very correct. Of course, you could declare such a function before that. The main improvement is that now it's possible to perform SAM conversion if such function is in Kotlin.

Reply:
✅